### PR TITLE
feat:  when a dimension is locked, disable selections on that dimension

### PR DESCRIPTION
**Pull request by LiKang6688**
- check qLock in each qDimensionInfo
- when a cell is locked or not a dimension, the selection should not work

### CODE REVIEW

**Comment by haxxmaxx** [comment]
put it on the cell as well, as we do with isDim

**Comment by LiKang6688** [comment]
> put it on the cell as well, as we do with isDim in handleData

Good point. Gonna do.

**Comment by haxxmaxx** [comment]
actually we only use isDim on the cell to determine if a cell is selectable or not, so this would be even cleaner
`isSelectable: c.isDim && !c.isLocked,`

File: src/table/components/TableBodyWrapper.jsx [hunk 1]
```diff
@@ -67,17 +67,19 @@ function TableBodyWrapper({
       {rows.map((row, rowIndex) => (
         <TableRow hover={hoverEffect} tabIndex={-1} key={row.key} sx={rowCellStyle} className="sn-table-row">
           {columns.map((column, columnIndex) => {
-            const cell = row[column.id];
+            const { id, align } = column;
```
Comment (haxxmaxx) [inline src/table/components/TableBodyWrapper.jsx:70]: I'd say this is a bit overkill but fine with me

File: src/table/utils/__tests__/handle-key-press.spec.js [hunk 2]
```diff
@@ -177,7 +177,7 @@ describe('handle-key-press', () => {
         cancel: jest.fn(),
         isModal: () => isModal,
       };
-      cell = { qElemNumber: 1, colIdx: 1, rowIdx: 1, isDim: true };
+      cell = { qElemNumber: 1, colIdx: 1, rowIdx: 1, isDim: true, isSelectable: true };
```
Comment (haxxmaxx) [inline src/table/utils/__tests__/handle-key-press.spec.js:180]: you can remove isDim here

File: src/table/utils/__tests__/handle-key-press.spec.js [hunk 3]
```diff
@@ -457,8 +457,8 @@ describe('handle-key-press', () => {
         announce,
         keyboard,
       });
-      expect(evt.preventDefault).not.toHaveBeenCalledTimes(1);
-      expect(evt.stopPropagation).not.toHaveBeenCalledTimes(1);
+      expect(evt.preventDefault).not.toHaveBeenCalled();
```
Comment (haxxmaxx) [inline src/table/utils/__tests__/handle-key-press.spec.js:460]: isn't this less specific?

**Review by haxxmaxx — APPROVED** [review]

File: src/table/utils/__tests__/handle-key-press.spec.js (hunk 3 above)
Comment (LiKang6688) [reply src/table/utils/__tests__/handle-key-press.spec.js:460]: if it calls two times, `not.toHaveBeenCalledTimes(1)` is still true, right?

File: src/table/utils/__tests__/handle-key-press.spec.js (hunk 2 above)
Comment (LiKang6688) [reply src/table/utils/__tests__/handle-key-press.spec.js:180]: right, gonna do

File: src/table/utils/__tests__/handle-key-press.spec.js (hunk 3 above)
Comment (haxxmaxx) [reply src/table/utils/__tests__/handle-key-press.spec.js:460]: ah sorry it is a `.not.` check, yes your change is better

**Pull request merged**